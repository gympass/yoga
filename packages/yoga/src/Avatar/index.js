import Avatar from './web/Avatar';
import AvatarCircle from './web/AvatarCircle';

const ExportAvatar = Object.assign(Avatar, { Circle: AvatarCircle });

export default ExportAvatar;
