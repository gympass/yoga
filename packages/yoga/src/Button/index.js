import Button from './web/Button';
import ButtonOutline from './web/Outline';
import ButtonText from './web/Text';
import ButtonLink from './web/Link';
import ButtonIcon from './web/Icon';

const ExportButton = Object.assign(Button, {
  Outline: ButtonOutline,
  Text: ButtonText,
  Link: ButtonLink,
  Icon: ButtonIcon,
});

export default ExportButton;
