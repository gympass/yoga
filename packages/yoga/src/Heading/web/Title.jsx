import React from 'react';
import PropTypes from 'prop-types';
import Text from '../../Text';
import Box from '../../Box';

const Title = ({ children, ...props }) => (
  <Box w="100%" textAlign="center">
    <Text.Medium {...props} size="small" as="h1">
      {children}
    </Text.Medium>
  </Box>
);

Title.propTypes = {
  children: PropTypes.string.isRequired,
};

Title.displayName = 'Heading.Title';

export default Title;
