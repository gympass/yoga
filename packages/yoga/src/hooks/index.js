export * from './usePortal';
export * from './useKeyPress';
export * from './useCombinedRefs';
export * from './useId';
