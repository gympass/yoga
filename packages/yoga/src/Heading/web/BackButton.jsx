import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';
import { ArrowLeft } from '@gympass/yoga-icons';
import Button from '../../Button';
import Box from '../../Box';
import Icon from '../../Icon';

const ButtonIcon = styled(Button)`
  ${({
    secondary,
    theme: {
      yoga: {
        components: { heading },
      },
    },
  }) => {
    return css`
      background-color: transparent;
      box-shadow: none;
      max-width: ${heading.button.width}px;
      min-width: ${heading.button.width}px;
      height: ${heading.button.height}px;

      &:active {
        background-color: transparent;
        & svg {
          fill: ${heading.button.color};
        }
      }

      &:not([disabled]):hover {
        background-color: transparent;
        box-shadow: none;
      }

      &:not([disabled]):hover {
        background-color: transparent;
        box-shadow: none;
      }
      :not([disabled]):focus {
        box-shadow: none;
      }

      & div {
        min-width: ${heading.button.backgroundWidth}px;
        max-width: ${heading.button.backgroundWidth}px;
        height: ${heading.button.backgroundHeight}px;
        background-color: ${secondary
          ? 'transparent'
          : heading.button.background};
      }

      & svg {
        fill: ${heading.button.color};
        margin: 0;
      }
    `;
  }};
`;

const BackButton = ({ onClick, ...props }) => (
  <ButtonIcon onClick={onClick} padding="xxxsmall" {...props}>
    <Box
      flex={1}
      display="flex"
      justifyContent="center"
      alignItems="center"
      borderRadius="50%"
      width="xxlarge"
      height="xxlarge"
    >
      <Icon as={ArrowLeft} role="img" />
    </Box>
  </ButtonIcon>
);

BackButton.propTypes = {
  onClick: PropTypes.func.isRequired,
};

BackButton.displayName = 'Heading.BackButton';

export default BackButton;
