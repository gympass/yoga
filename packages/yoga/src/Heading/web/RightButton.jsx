import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';
import Button from '../../Button';
import Icon from '../../Icon';
import Box from '../../Box';

const ButtonIcon = styled(Button)`
  ${({
    secondary,
    theme: {
      yoga: {
        components: { heading },
      },
    },
  }) => {
    return css`
      background-color: transparent;
      box-shadow: none;
      max-width: ${heading.button.width}px;
      min-width: ${heading.button.width}px;
      height: ${heading.button.height}px;

      &:active {
        background-color: transparent;
        & svg {
          fill: ${heading.button.color};
        }
      }

      &:not([disabled]):hover {
        background-color: transparent;
        box-shadow: none;
      }

      &:not([disabled]):hover {
        background-color: transparent;
        box-shadow: none;
      }
      :not([disabled]):focus {
        box-shadow: none;
      }

      & div {
        min-width: ${heading.button.backgroundWidth}px;
        max-width: ${heading.button.backgroundWidth}px;
        height: ${heading.button.backgroundHeight}px;
        background-color: ${secondary
          ? 'transparent'
          : heading.button.background};
      }

      & svg {
        fill: ${heading.button.color};
        margin: 0;
      }
    `;
  }};
`;

const RightButton = ({ onClick, icon, ...props }) => (
  <ButtonIcon onClick={onClick} secondary padding="xxxsmall" {...props}>
    <Box
      flex={1}
      display="flex"
      justifyContent="center"
      alignItems="center"
      borderRadius="50%"
      width="xxlarge"
      height="xxlarge"
    >
      <Icon as={icon} role="img" />
    </Box>
  </ButtonIcon>
);

RightButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  icon: PropTypes.elementType.isRequired,
};

RightButton.displayName = 'Heading.RightButton';

export default RightButton;
