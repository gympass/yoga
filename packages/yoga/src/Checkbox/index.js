import { Checkbox, Switch } from './web';

const ExportCheckbox = Object.assign(Checkbox, {
  Switch,
});

export default ExportCheckbox;
