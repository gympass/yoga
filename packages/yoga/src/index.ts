import ThemeProvider, {
  FontLoader,
  yogaTheme,
  theme,
  createTheme,
  v3theme,
  type Theme,
} from './Theme';
import Accordion from './Accordion';
import Button from './Button';
import List from './List';
import Checkbox from './Checkbox';
import Slider from './Slider';
import RadioGroup from './RadioGroup';
import { Card, PlanCard, EventCard } from './Card';
import Stepper from './Stepper';
import { Container, Row, Col, Hide } from './Grid';
import Text from './Text';
import Rating from './Rating';
import Tag from './Tag';
import Input from './Input';
import Progress from './Progress';
import Dropdown from './Dropdown';
import TextArea from './TextArea';
import AutoComplete from './AutoComplete';
import Icon from './Icon';
import Chips from './Chips';
import Box from './Box';
import Snackbar from './Snackbar';
import Avatar from './Avatar';
import BottomSheet from './BottomSheet';
import Dialog from './Dialog';
import Divider from './Divider';
import Header from './Header';
import Menu from './Menu';
import Banner from './Banner';
import Heading from './Heading';
import Feedback from './Feedback';
import Datepicker from './Datepicker';
import Drawer from './Drawer';
import Skeleton from './Skeleton';
import ActionRequirement from './ActionRequirement';
import Popover from './Popover';
import Spinner from './Spinner';
import NavigationMenu from './NavigationMenu';

export {
  Accordion,
  ThemeProvider,
  FontLoader,
  createTheme,
  yogaTheme,
  theme,
  v3theme,
  Button,
  List,
  Checkbox,
  RadioGroup,
  Slider,
  Card,
  PlanCard,
  Stepper,
  Container,
  Row,
  Col,
  Hide,
  Text,
  Rating,
  EventCard,
  Tag,
  Input,
  Progress,
  Dropdown,
  TextArea,
  AutoComplete,
  Icon,
  Chips,
  Box,
  Snackbar,
  Avatar,
  BottomSheet,
  Dialog,
  Divider,
  Header,
  Menu,
  Banner,
  Heading,
  Feedback,
  Datepicker,
  Drawer,
  Skeleton,
  ActionRequirement,
  Popover,
  Spinner,
  NavigationMenu,
  Theme,
};
