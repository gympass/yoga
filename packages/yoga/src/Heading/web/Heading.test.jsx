import React from 'react';
import { render } from '@testing-library/react';

import { Upload2 } from '@gympass/yoga-icons/src';
import { ThemeProvider, Heading } from '../..';
import Title from './Title';
import BackButton from './BackButton';
import RightButton from './RightButton';

const onClick = () => {};

describe('<Heading />', () => {
  it('should match snapshot with title', () => {
    const { container } = render(
      <ThemeProvider>
        <Heading>
          <Title>Gympass</Title>
        </Heading>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with title and back button', () => {
    const { container } = render(
      <ThemeProvider>
        <Heading>
          <Title>Gympass</Title>
          <BackButton onClick={onClick} />
        </Heading>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with all components', () => {
    const { container } = render(
      <ThemeProvider>
        <Heading>
          <Title>Gympass</Title>
          <BackButton onClick={onClick} />
          <RightButton icon={Upload2} onClick={onClick} />
        </Heading>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with back button', () => {
    const { container } = render(
      <ThemeProvider>
        <Heading>
          <BackButton onClick={onClick} />
        </Heading>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot no padding', () => {
    const { container } = render(
      <ThemeProvider>
        <Heading noPadding>
          <Title>Gympass</Title>
          <BackButton onClick={onClick} />
          <RightButton icon={Upload2} onClick={onClick} />
        </Heading>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should have aria-label', () => {
    const { getByLabelText } = render(
      <ThemeProvider>
        <Heading noPadding>
          <Title>Gympass</Title>
          <BackButton onClick={onClick} />
          <RightButton
            icon={Upload2}
            onClick={onClick}
            aria-label="labelAriaText"
          />
        </Heading>
      </ThemeProvider>,
    );

    expect(getByLabelText('labelAriaText')).toBeTruthy();
  });

  it('should override the background color', () => {
    const { container } = render(
      <ThemeProvider>
        <Heading bg="yoga">
          <Title>Gympass</Title>
        </Heading>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
