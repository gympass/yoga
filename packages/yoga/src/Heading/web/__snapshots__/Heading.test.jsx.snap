// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should match snapshot no padding 1`] = `
.c5 {
  width: 100%;
  text-align: center;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  width: 12px;
  height: 12px;
}

.c1 {
  box-sizing: border-box;
  text-align: center;
  outline: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  height: 48px;
  padding-left: 24px;
  padding-right: 24px;
  background-color: #D8385E;
  border: none;
  border-radius: 9999px;
  color: #FFFFFF;
  font-size: 16px;
  font-weight: 500;
  font-family: Rubik;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 svg {
  width: 24px;
  height: 24px;
  fill: #FFFFFF;
  margin-right: 8px;
  -webkit-transition: fill 0.2s;
  transition: fill 0.2s;
}

.c1:not([disabled]):hover,
.c1:not([disabled]):focus {
  box-shadow: 0 4px 8px rgba(216,56,94,0.45);
}

.c1:active {
  background-color: rgba(216,56,94,0.75);
  color: #FFFFFF;
}

.c1:active svg {
  fill: #FFFFFF;
}

.c8 {
  box-sizing: border-box;
  text-align: center;
  outline: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  height: 48px;
  padding-left: 24px;
  padding-right: 24px;
  background-color: #231B22;
  border: none;
  border-radius: 9999px;
  color: #FFFFFF;
  font-size: 16px;
  font-weight: 500;
  font-family: Rubik;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c8 svg {
  width: 24px;
  height: 24px;
  fill: #FFFFFF;
  margin-right: 8px;
  -webkit-transition: fill 0.2s;
  transition: fill 0.2s;
}

.c8:not([disabled]):hover,
.c8:not([disabled]):focus {
  box-shadow: 0 4px 8px rgba(35,27,34,0.45);
}

.c8:active {
  background-color: rgba(35,27,34,0.75);
  color: #FFFFFF;
}

.c8:active svg {
  fill: #FFFFFF;
}

.c6 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 500;
  font-size: 14px;
}

.c0 {
  background: #FFFFFF;
  padding: 4px 16px;
  min-height: 56px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0;
}

.c2 {
  background-color: transparent;
  box-shadow: none;
  max-width: 48px;
  min-width: 48px;
  height: 48px;
}

.c2:active {
  background-color: transparent;
}

.c2:active svg {
  fill: #231B22;
}

.c2:not([disabled]):hover {
  background-color: transparent;
  box-shadow: none;
}

.c2:not([disabled]):hover {
  background-color: transparent;
  box-shadow: none;
}

.c2:not([disabled]):focus {
  box-shadow: none;
}

.c2 div {
  min-width: 40px;
  max-width: 40px;
  height: 40px;
  background-color: #D7D7E0;
}

.c2 svg {
  fill: #231B22;
  margin: 0;
}

.c9 {
  background-color: transparent;
  box-shadow: none;
  max-width: 48px;
  min-width: 48px;
  height: 48px;
}

.c9:active {
  background-color: transparent;
}

.c9:active svg {
  fill: #231B22;
}

.c9:not([disabled]):hover {
  background-color: transparent;
  box-shadow: none;
}

.c9:not([disabled]):hover {
  background-color: transparent;
  box-shadow: none;
}

.c9:not([disabled]):focus {
  box-shadow: none;
}

.c9 div {
  min-width: 40px;
  max-width: 40px;
  height: 40px;
  background-color: transparent;
}

.c9 svg {
  fill: #231B22;
  margin: 0;
}

<div>
  <header
    class="c0"
  >
    <button
      class="c1 c2"
    >
      <div
        class="c3"
        display="flex"
        height="xxlarge"
        width="xxlarge"
      >
        <svg
          aria-hidden="true"
          class="c4"
          height="12"
          role="img"
          width="12"
        />
      </div>
    </button>
    <div
      class="c5"
    >
      <h1
        class="c6"
        font-size="small"
      >
        Gympass
      </h1>
    </div>
    <div
      class="c7"
      display="flex"
    >
      <button
        class="c8 c9"
      >
        <div
          class="c3"
          display="flex"
          height="xxlarge"
          width="xxlarge"
        >
          <svg
            aria-hidden="true"
            class="c4"
            height="12"
            role="img"
            width="12"
          />
        </div>
      </button>
    </div>
  </header>
</div>
`;

exports[`<Heading /> should match snapshot with all components 1`] = `
.c5 {
  width: 100%;
  text-align: center;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  width: 12px;
  height: 12px;
}

.c1 {
  box-sizing: border-box;
  text-align: center;
  outline: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  height: 48px;
  padding-left: 24px;
  padding-right: 24px;
  background-color: #D8385E;
  border: none;
  border-radius: 9999px;
  color: #FFFFFF;
  font-size: 16px;
  font-weight: 500;
  font-family: Rubik;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 svg {
  width: 24px;
  height: 24px;
  fill: #FFFFFF;
  margin-right: 8px;
  -webkit-transition: fill 0.2s;
  transition: fill 0.2s;
}

.c1:not([disabled]):hover,
.c1:not([disabled]):focus {
  box-shadow: 0 4px 8px rgba(216,56,94,0.45);
}

.c1:active {
  background-color: rgba(216,56,94,0.75);
  color: #FFFFFF;
}

.c1:active svg {
  fill: #FFFFFF;
}

.c8 {
  box-sizing: border-box;
  text-align: center;
  outline: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  height: 48px;
  padding-left: 24px;
  padding-right: 24px;
  background-color: #231B22;
  border: none;
  border-radius: 9999px;
  color: #FFFFFF;
  font-size: 16px;
  font-weight: 500;
  font-family: Rubik;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c8 svg {
  width: 24px;
  height: 24px;
  fill: #FFFFFF;
  margin-right: 8px;
  -webkit-transition: fill 0.2s;
  transition: fill 0.2s;
}

.c8:not([disabled]):hover,
.c8:not([disabled]):focus {
  box-shadow: 0 4px 8px rgba(35,27,34,0.45);
}

.c8:active {
  background-color: rgba(35,27,34,0.75);
  color: #FFFFFF;
}

.c8:active svg {
  fill: #FFFFFF;
}

.c6 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 500;
  font-size: 14px;
}

.c0 {
  background: #FFFFFF;
  padding: 4px 16px;
  min-height: 56px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  background-color: transparent;
  box-shadow: none;
  max-width: 48px;
  min-width: 48px;
  height: 48px;
}

.c2:active {
  background-color: transparent;
}

.c2:active svg {
  fill: #231B22;
}

.c2:not([disabled]):hover {
  background-color: transparent;
  box-shadow: none;
}

.c2:not([disabled]):hover {
  background-color: transparent;
  box-shadow: none;
}

.c2:not([disabled]):focus {
  box-shadow: none;
}

.c2 div {
  min-width: 40px;
  max-width: 40px;
  height: 40px;
  background-color: #D7D7E0;
}

.c2 svg {
  fill: #231B22;
  margin: 0;
}

.c9 {
  background-color: transparent;
  box-shadow: none;
  max-width: 48px;
  min-width: 48px;
  height: 48px;
}

.c9:active {
  background-color: transparent;
}

.c9:active svg {
  fill: #231B22;
}

.c9:not([disabled]):hover {
  background-color: transparent;
  box-shadow: none;
}

.c9:not([disabled]):hover {
  background-color: transparent;
  box-shadow: none;
}

.c9:not([disabled]):focus {
  box-shadow: none;
}

.c9 div {
  min-width: 40px;
  max-width: 40px;
  height: 40px;
  background-color: transparent;
}

.c9 svg {
  fill: #231B22;
  margin: 0;
}

@media (min-width:769px) {
  .c0 {
    padding: 4px 16px;
  }
}

<div>
  <header
    class="c0"
  >
    <button
      class="c1 c2"
    >
      <div
        class="c3"
        display="flex"
        height="xxlarge"
        width="xxlarge"
      >
        <svg
          aria-hidden="true"
          class="c4"
          height="12"
          role="img"
          width="12"
        />
      </div>
    </button>
    <div
      class="c5"
    >
      <h1
        class="c6"
        font-size="small"
      >
        Gympass
      </h1>
    </div>
    <div
      class="c7"
      display="flex"
    >
      <button
        class="c8 c9"
      >
        <div
          class="c3"
          display="flex"
          height="xxlarge"
          width="xxlarge"
        >
          <svg
            aria-hidden="true"
            class="c4"
            height="12"
            role="img"
            width="12"
          />
        </div>
      </button>
    </div>
  </header>
</div>
`;

exports[`<Heading /> should match snapshot with back button 1`] = `
.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  width: 12px;
  height: 12px;
}

.c5 {
  width: 100%;
}

.c1 {
  box-sizing: border-box;
  text-align: center;
  outline: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  height: 48px;
  padding-left: 24px;
  padding-right: 24px;
  background-color: #D8385E;
  border: none;
  border-radius: 9999px;
  color: #FFFFFF;
  font-size: 16px;
  font-weight: 500;
  font-family: Rubik;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 svg {
  width: 24px;
  height: 24px;
  fill: #FFFFFF;
  margin-right: 8px;
  -webkit-transition: fill 0.2s;
  transition: fill 0.2s;
}

.c1:not([disabled]):hover,
.c1:not([disabled]):focus {
  box-shadow: 0 4px 8px rgba(216,56,94,0.45);
}

.c1:active {
  background-color: rgba(216,56,94,0.75);
  color: #FFFFFF;
}

.c1:active svg {
  fill: #FFFFFF;
}

.c0 {
  background: #FFFFFF;
  padding: 4px 16px;
  min-height: 56px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  width: 48px;
  height: 48px;
}

.c2 {
  background-color: transparent;
  box-shadow: none;
  max-width: 48px;
  min-width: 48px;
  height: 48px;
}

.c2:active {
  background-color: transparent;
}

.c2:active svg {
  fill: #231B22;
}

.c2:not([disabled]):hover {
  background-color: transparent;
  box-shadow: none;
}

.c2:not([disabled]):hover {
  background-color: transparent;
  box-shadow: none;
}

.c2:not([disabled]):focus {
  box-shadow: none;
}

.c2 div {
  min-width: 40px;
  max-width: 40px;
  height: 40px;
  background-color: #D7D7E0;
}

.c2 svg {
  fill: #231B22;
  margin: 0;
}

@media (min-width:769px) {
  .c0 {
    padding: 4px 16px;
  }
}

<div>
  <header
    class="c0"
  >
    <button
      class="c1 c2"
    >
      <div
        class="c3"
        display="flex"
        height="xxlarge"
        width="xxlarge"
      >
        <svg
          aria-hidden="true"
          class="c4"
          height="12"
          role="img"
          width="12"
        />
      </div>
    </button>
    <div
      class="c5"
    />
    <div
      class="c6"
      display="flex"
    >
      <div
        class="c7"
      />
    </div>
  </header>
</div>
`;

exports[`<Heading /> should match snapshot with title 1`] = `
.c2 {
  width: 100%;
  text-align: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 500;
  font-size: 14px;
}

.c0 {
  background: #FFFFFF;
  padding: 4px 16px;
  min-height: 56px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  width: 48px;
  height: 48px;
}

@media (min-width:769px) {
  .c0 {
    padding: 4px 16px;
  }
}

<div>
  <header
    class="c0"
  >
    <div
      class="c1"
    />
    <div
      class="c2"
    >
      <h1
        class="c3"
        font-size="small"
      >
        Gympass
      </h1>
    </div>
    <div
      class="c4"
      display="flex"
    >
      <div
        class="c1"
      />
    </div>
  </header>
</div>
`;

exports[`<Heading /> should match snapshot with title and back button 1`] = `
.c5 {
  width: 100%;
  text-align: center;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  border-radius: 50%;
  width: 40px;
  height: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  width: 12px;
  height: 12px;
}

.c1 {
  box-sizing: border-box;
  text-align: center;
  outline: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  height: 48px;
  padding-left: 24px;
  padding-right: 24px;
  background-color: #D8385E;
  border: none;
  border-radius: 9999px;
  color: #FFFFFF;
  font-size: 16px;
  font-weight: 500;
  font-family: Rubik;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 svg {
  width: 24px;
  height: 24px;
  fill: #FFFFFF;
  margin-right: 8px;
  -webkit-transition: fill 0.2s;
  transition: fill 0.2s;
}

.c1:not([disabled]):hover,
.c1:not([disabled]):focus {
  box-shadow: 0 4px 8px rgba(216,56,94,0.45);
}

.c1:active {
  background-color: rgba(216,56,94,0.75);
  color: #FFFFFF;
}

.c1:active svg {
  fill: #FFFFFF;
}

.c6 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 500;
  font-size: 14px;
}

.c0 {
  background: #FFFFFF;
  padding: 4px 16px;
  min-height: 56px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 {
  width: 48px;
  height: 48px;
}

.c2 {
  background-color: transparent;
  box-shadow: none;
  max-width: 48px;
  min-width: 48px;
  height: 48px;
}

.c2:active {
  background-color: transparent;
}

.c2:active svg {
  fill: #231B22;
}

.c2:not([disabled]):hover {
  background-color: transparent;
  box-shadow: none;
}

.c2:not([disabled]):hover {
  background-color: transparent;
  box-shadow: none;
}

.c2:not([disabled]):focus {
  box-shadow: none;
}

.c2 div {
  min-width: 40px;
  max-width: 40px;
  height: 40px;
  background-color: #D7D7E0;
}

.c2 svg {
  fill: #231B22;
  margin: 0;
}

@media (min-width:769px) {
  .c0 {
    padding: 4px 16px;
  }
}

<div>
  <header
    class="c0"
  >
    <button
      class="c1 c2"
    >
      <div
        class="c3"
        display="flex"
        height="xxlarge"
        width="xxlarge"
      >
        <svg
          aria-hidden="true"
          class="c4"
          height="12"
          role="img"
          width="12"
        />
      </div>
    </button>
    <div
      class="c5"
    >
      <h1
        class="c6"
        font-size="small"
      >
        Gympass
      </h1>
    </div>
    <div
      class="c7"
      display="flex"
    >
      <div
        class="c8"
      />
    </div>
  </header>
</div>
`;
